### PR TITLE
MODQM-11 - marc-json <-> ui-compatible-marc-json converter updating

### PR DESCRIPTION
**Pull request by siarhei-charniak**
[MODQM-11](https://issues.folio.org/browse/MODQM-11) - marc-json <-> ui-compatible-marc-json converter updating

## Purpose
QuickMarcJson <-> ParsedRecord converters should be updated to support the ability to represent the 006 and 007 control fields broken out by byte (like the 008 control field)

## Approach
* Both converters are updated to support 006, 007 control fields broken out by byte representation
* quick_marc_json.sample is updated

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [x] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/java/org/folio/util/ErrorUtils.java [hunk 1]
```diff
@@ -37,6 +38,9 @@ public static Error buildError(int status, ErrorType type, String message) {
         status = ((HttpException) cause).getCode();
         JsonObject errorJsonObject = ((HttpException) cause).getError();
         error = resolveError(errorJsonObject);
+      } else if (cause instanceof IllegalArgumentException) {
```
Comment (khandramai) [inline src/main/java/org/folio/util/ErrorUtils.java:41]: Is it make sense to introduce custom exception, for example, ConverterException, for all thrown exceptions from Converters?

Comment (siarhei-charniak) [reply src/main/java/org/folio/util/ErrorUtils.java]: introduced MarcConversionException

File: src/test/java/org/folio/rest/impl/QuickMarcApiTest.java [hunk 2]
```diff
@@ -121,8 +121,11 @@ void testUpdateQuickMarcRecordWrongUuid() {
 
     wireMockServer
       .stubFor(put(urlEqualTo(getResourceByIdPath(CM_RECORDS, wrongUUID)))
-        .withRequestBody(containing(getJsonObject(PARSED_RECORD_DTO_PATH).encode()))
-        .willReturn(aResponse().withStatus(404)));
+//        .withRequestBody(containing(getJsonObject(PARSED_RECORD_DTO_PATH).encode()))
```
Comment (khandramai) [inline src/test/java/org/folio/rest/impl/QuickMarcApiTest.java:124]: Please, remove commented line.

Comment (siarhei-charniak) [reply src/test/java/org/folio/rest/impl/QuickMarcApiTest.java]: removed

**Pull request merged**